py-server/main.py

```diff
@@ -60,18 +66,7 @@ def chat():
     message_history_json = flask.request.args.get(
         "message_history"
     ) or flask.request.form.get("message_history")
-    message_history = None
-    try:
-        message_history = [
-            ChatMessage(content=msg["content"], author=msg["author"])
-            for msg in json.loads(message_history_json)
-        ]
-        print("message_history", message_history)
-        if not message_history:
-            # Don't send an empty message history list
-            message_history = None
-    except Exception as e:
-        print("failed to parse chat history", e)
+    message_history = build_message_history(message_history_json)
```

Comment:
re-introduce try/except block

Reply:
I think it's better to throw than to swallow the exception though, since it wasn't obvious it was quietly dropping the chat history.

I catch Exception in build_message_history now, and then throw a ValueError instead so the server will return a 304.

The client doesn't deal with this right now, but we should probably add that in a follow up.

Reply:
agree about not swallowing the exception, though 304 doesn't sound like the right status code. can we respond with a 400 bad request? (fine to leave for a follow up)